tests/unittests/tests-hashes/tests-hashes-sha256-hmac.c

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright (C) 2016 Martin Landsmann <martin.landsmann@haw-hamburg.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+#include <limits.h>
+#include <string.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "embUnit/embUnit.h"
+
+#include "hashes/sha256.h"
+
+#include "tests-hashes.h"
+
+static int compare_str_vs_digest(const char *str,
+                                 const unsigned char hash[SHA256_DIGEST_LENGTH])
+{
+    char ch[3] = { 0, 0, 0 };
+    size_t iter_hash = 0;
+    size_t str_length = strlen(str);
+    for (size_t i = 0; i < str_length; i += 2) {
+        ch[0] = str[i];
+        ch[1] = str[i + 1];
+
+        if (hash[iter_hash++] != strtol(ch, NULL, 16)) {
+            return 0;
+        }
+    }
+    return 1;
+}
+
+static void test_hashes_hmac_sha256_hash_sequence(void)
+{
+    unsigned char key[64];
```

Comment:
make this var `static`, the function will not be thread safe, but I don't think it is going to be called from multiple threads simultaneously, and there is a static buffer below as well so there's that too..